### PR TITLE
Use composition and service decoration for router

### LINKED ISSUE #20: The special prefix_except_default router should use composition instead of inheritance
The custom router current uses inheritance, but it only wraps public methods. This means that it could totally be implemented using composition instead.

This would even simplify things, by avoiding to maintain the dependencies of the parent router in your constructor, and having to mix your own options with the Symfony ones.

### PR DESCRIPTION
**Pull request by kbond**
Fixes #20 

This combined with #18 with remove the need for the compiler pass.

### CODE REVIEW

**Comment by kbond** [comment]
<del>I still need to get the specs fixed.</del> Fixed.

File: src/Routing/Router.php [hunk 1]
```diff
@@ -2,72 +2,92 @@
...
-use InvalidArgumentException;
+use Symfony\Component\Routing\RouterInterface;
 
-class Router extends BaseRouter
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class Router implements RouterInterface, RequestMatcherInterface, WarmableInterface
 {
+    private $innerRouter;
```
Comment (umpirsky) [inline src/Routing/Router.php]: `private $router;` is enough. ;)

**Review by umpirsky — APPROVED** [review]
Looks great, thanks!

File: src/Routing/Router.php (hunk 1 above)
Comment (kbond) [reply src/Routing/Router.php]: Ok, will update

**Comment by kbond** [comment]
Now that #18 is merged I will remove the compiler pass

**Comment by kbond** [comment]
CompilerPass removed and rebased.

**Comment by umpirsky** [comment]
Perfect, we should tag a new release once we merge all.

Thanks @kbond, really appreciate huge effort! 🥇

**Comment by kbond** [comment]
How about that new release? ;)

**Comment by umpirsky** [comment]
@kbond Done, thanks!

**Pull request merged**